embedded/src/main.rs

```diff
@@ -0,0 +1,76 @@
+#![feature(alloc_error_handler)]
+#![feature(panic_info_message)]
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+extern crate bitcoin;
+
+use alloc::string::ToString;
+use alloc::vec;
+use core::alloc::Layout;
+use core::panic::PanicInfo;
+
+use alloc_cortex_m::CortexMHeap;
+// use panic_halt as _;
+use bitcoin::{Address, Network, PrivateKey};
+use bitcoin::secp256k1::ffi::types::AlignedType;
+use bitcoin::secp256k1::Secp256k1;
+
+use cortex_m::asm;
+use cortex_m_rt::entry;
+use cortex_m_semihosting::{debug, hprintln};
+
+// this is the allocator the application will use
+#[global_allocator]
+static ALLOCATOR: CortexMHeap = CortexMHeap::empty();
+
+const HEAP_SIZE: usize = 1024 * 512; // 512 KB
```

Comment:
all memory available is 512Kb according to `memory.x` so the heap should be less, eg 256KB

Reply:
Will fix

Reply:
Probably not worth to dismiss approval for this